lib/ansible/plugins/inventory/foreman.py

```diff
@@ -121,134 +126,134 @@ def _get_session(self):
             self.session.verify = self.get_option('validate_certs')
         return self.session
 
-    def _get_json(self, url, ignore_errors=None):
+    def _get_all_hosts(self):
 
-        if not self.use_cache or url not in self._cache.get(self.cache_key, {}):
+        if self.use_cache:
+            inventory = self._cache[self.cache_key]
```

Comment:
```suggestion
            try:
                inventory = self._cache[self.cache_key]
            except KeyError:
                # The cache key expired or does not exist yet
                inventory = None
```

Reply:
Since caching was working in the integration tests before you shouldn't need to modify that aspect of this inventory plugin.